src/logic/wallets/pairing/hooks/useGetPairingUri.ts

```diff
@@ -0,0 +1,13 @@
+import { getPairingUri } from 'src/logic/wallets/pairing/utils'
+import { useEffect, useState } from 'react'
+
+export const useGetPairingUri = (): string | undefined => {
+  const onboardUri = getPairingUri()
+  const [uri, setUri] = useState<string>()
+
+  useEffect(() => {
+    setUri(onboardUri)
+  }, [onboardUri])
```

Comment:
How's this different from just calling `getPairingUri()` from the PairingDetails component?

Reply:
Functionally, it is the same but just hiding the complexity in a custom hook in line with the other onboard hooks we have. Keeps `PairingDetails` a bit shorter/readable. In both cases we need a component state to ensure that `onboard` calls are integrated into the render cycle of `PairingDetails`. For later refactoring we should move `onboard` into a global context within our React App instead of managing it in a file outside of React.